dask/dataframe/groupby.py

```diff
@@ -1742,15 +1741,12 @@ def _shuffle(self, meta):
 
         if isinstance(self.by, DataFrame):  # add by columns to dataframe
             df2 = df.assign(**{"_by_" + c: self.by[c] for c in self.by.columns})
-            by = self.by
         elif isinstance(self.by, Series):
             df2 = df.assign(_by=self.by)
-            by = self.by
         else:
             df2 = df
-            by = df._select_columns_or_index(self.by)
 
-        df3 = shuffle(df2, by)  # shuffle dataframe and index
+        df3 = df2.shuffle(on=self.by)  # shuffle dataframe and index
```

Comment:
Note that we don't actually **want** this `self.by` validation, because `shuffle` is already designed to do this validation better.  I say "better", because it will be able to reduce the overall memory usage of a task-based shuffle when column names are specified (without affecting p2p performance).

Reply:
I don't quite understand this comment

> because shuffle is already designed to do this better

Can you point me to where this is happening?

Reply:
This is the line that checks if we are passing in column names to shuffle on: https://github.com/dask/dask/blob/79960cf744f4c7ab048924bab404df06e4e5e4a9/dask/dataframe/shuffle.py#L374

If we enter that `if` block,  `rearrange_by_column` will just use the hash dispatch on those columns instead of looking for the `"_partitions"` column (which would have taken extra memory to create and pass around).

If we dont enter that `if` block, `shuffle` [will still call `_select_columns_or_index`](https://github.com/dask/dask/blob/79960cf744f4c7ab048924bab404df06e4e5e4a9/dask/dataframe/shuffle.py#L398) if necessary.